vendor/bat-native-ledger/src/bat/ledger/internal/common/bind_util.cc

```diff
@@ -543,4 +544,144 @@ ledger::MonthlyReportInfoPtr FromStringToMonthlyReport(
+    order_item.SetDoubleKey("price", item->price);
+    order_item.SetStringKey("name", item->name);
+    order_item.SetStringKey("description", item->description);
+    order_item.SetIntKey("type", static_cast<int>(item->type));
+    order_item.SetStringKey("expires_at", std::to_string(item->expires_at));
+    items.Append(std::move(order_item));
+  }
+
+  base::Value order(base::Value::Type::DICTIONARY);
+  order.SetStringKey("order_id", info->order_id);
+  order.SetDoubleKey("total_amount", info->total_amount);
+  order.SetStringKey("merchant_id", info->merchant_id);
+  order.SetStringKey("location", info->location);
+  order.SetIntKey("status", static_cast<int>(info->status));
+  order.SetStringKey("contribution_id", info->contribution_id);
+  order.SetKey("items", std::move(items));
+
+  std::string json;
+  base::JSONWriter::Write(order, &json);
+
+  return json;
+}
+
+ledger::SKUOrderPtr FromStringToSKUOrder(const std::string& data) {
+  base::Optional<base::Value> value = base::JSONReader::Read(data);
+
+  if (!value || !value->is_dict()) {
+    return nullptr;
+  }
+
+  base::DictionaryValue* dictionary = nullptr;
+  if (!value->GetAsDictionary(&dictionary)) {
+    return nullptr;
+  }
+
+  auto order = ledger::SKUOrder::New();
+
+  std::string id;
+  const auto* order_id = dictionary->FindStringKey("order_id");
+  if (order_id) {
+    order->order_id = *order_id;
+    id = order->order_id;
+  }
+
+  if (id.empty()) {
+    return nullptr;
+  }
+
+  const auto total_amount = dictionary->FindDoubleKey("total_amount");
+  if (total_amount) {
+    order->total_amount = *total_amount;
+  }
+
+  const auto* merchant_id = dictionary->FindStringKey("merchant_id");
+  if (merchant_id) {
+    order->merchant_id = *merchant_id;
+  }
+
+  const auto* location = dictionary->FindStringKey("location");
+  if (location) {
+    order->location = *location;
+  }
+
+  const auto status = dictionary->FindIntKey("status");
+  if (status) {
+    order->status = static_cast<ledger::SKUOrderStatus>(*status);
+  }
+
+  const auto* contribution_id = dictionary->FindStringKey("contribution_id");
+  if (contribution_id) {
+    order->contribution_id = *contribution_id;
+  }
+
+  auto* items = dictionary->FindListKey("items");
+  if (items) {
+    ledger::SKUOrderItemPtr order_item = nullptr;
+    for (const auto& item : items->GetList()) {
+      order_item = ledger::SKUOrderItem::New();
+
+      const auto* order_item_id = item.FindStringKey("order_item_id");
+      if (order_item_id) {
+        order_item->order_item_id = *order_item_id;
+      }
+
+      order_item->order_id = id;
+
+      const auto* sku = item.FindStringKey("sku");
+      if (sku) {
+        order_item->sku = *sku;
+      }
+
+      const auto quantity = item.FindIntKey("quantity");
+      if (quantity) {
+        order_item->quantity = *quantity;
+      }
+
+      const auto price = item.FindDoubleKey("price");
+      if (price) {
+        order_item->price = *price;
+      }
```

Comment:
nit: `order_time->price = price.value_or(0.0);` is easier  to read as you can see the default value and no need to check for `nullptr`

Reply:
created an issue so that we can do it in one go for all instances https://github.com/brave/brave-browser/issues/9330